From a835c409a19a9c72b5e669a70664ea960d873704 Mon Sep 17 00:00:00 2001
From: Renata Ravanelli <rravanel@redhat.com>
Date: Tue, 12 Nov 2024 15:12:52 -0300
Subject: [PATCH 3/5] util/osrelease.py: improve whitespace and quote stripping

- Enhanced the value stripping logic in osrelease parsing
to handle leading and trailing spaces, newlines, tabs,
and both single and double quotes.
- This ensures cleaner and more accurate key-value assignments.

Signed-off-by: Renata Ravanelli <rravanel@redhat.com>
(cherry picked from commit 066f1ea89fbda6e886a5d88119586c0f09b0a234)
---
 osbuild/util/osrelease.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/osbuild/util/osrelease.py b/osbuild/util/osrelease.py
index b8d56e73..a2b61d26 100644
--- a/osbuild/util/osrelease.py
+++ b/osbuild/util/osrelease.py
@@ -33,7 +33,7 @@ def parse_files(*paths):
                 if line[0] == "#":
                     continue
                 key, value = line.split("=", 1)
-                osrelease[key] = value.strip('"')
+                osrelease[key] = value.strip(" \n\t'\"")
 
     return osrelease
 
-- 
2.47.0

